posix.mak

```diff
@@ -495,37 +495,38 @@ checkwhitespace: $(LIB)
 	make -C ../dscanner githash debug
 
 style: ../dscanner/dsc $(LIB)
-	@echo "Check for trailing whitespace"
-	grep -nr '[[:blank:]]$$' etc std ; test $$? -eq 1
+	@grep -nr '[[:blank:]]$$' etc std && \
+		{ echo "Check for trailing whitespace" && exit 1; } ; exit 0
```

Comment:
Cool, one simpler pattern that would work is: `pipe && { echo ... && return 1 } || return 0;`

The shell evaluates all `&&` and `||` indiscriminately left to right. Eh, I guess not much of a simplification, up to you.

Also I'm using return instead of exit because I'm under the delusion one day we may use https://www.gnu.org/software/make/manual/html_node/One-Shell.html